### ISSUE TITLE
emacs-mac build now expecting X toolkit

### ISSUE THREAD

**Issue by cwhatley** [railwaycat/homebrew-emacsmacport#247]
Just did a straight build of `brew install emacs-mac` on Big Sur. This is the first time I built since 27.2 was released. I tried installing libxaw and rebuilding and it appears to complete, but the app bundle is not launchable - the MacOS folder within contained only Emacs.sh and none of the built binaries.

```
==> ./configure --enable-locallisppath=/usr/local/share/emacs/site-lisp --infodir=/usr/local/Cellar/emacs-mac/emacs-27.2-mac-8.2/share/info/emacs --pref
Last 15 lines from /Users/cwhatley/Library/Logs/Homebrew/emacs-mac/02.configure:
checking for gio-2.0 >= 2.26... no
checking for gconf-2.0 >= 2.13... no
checking for lgetfilecon in -lselinux... no
checking for gnutls >= 2.12.2... yes
checking for libsystemd >= 222... no
checking for jansson >= 2.7... yes
checking for sys/inotify.h... no
checking for libkqueue... no
checking for library containing kqueue... none required
checking for xaw3d... no
checking for libXaw... configure: error: No X toolkit could be found.
If you are sure you want Emacs compiled without an X toolkit, pass
  --with-x-toolkit=no
to configure.  Otherwise, install the development libraries for the toolkit
that you want to use (e.g. Gtk+) and re-run configure.
```

**Comment by railwaycat** [comment]
Strange..emacs-mac should not has any dependency to any X toolkit. 

Could you attach the `00.options.out` and `02.configure` files under `~/Library/Logs/Homebrew/emacs-mac/` directory? From that point we will able to see what exactly happened on the configuration process.

**Comment by cwhatley** [comment]
The homebrew build is not detecting Carbon/Carbon.h.

I've run install of emacs-mac with --debug and then dropped into a shell to retry running configure by hand and it fails, but if I fire up my own shell without brew's shim magic and run configure in the same build folder, it works just fine.

[00.options.out.txt](https://github.com/railwaycat/homebrew-emacsmacport/files/6265157/00.options.out.txt)
[02.configure.cc.txt](https://github.com/railwaycat/homebrew-emacsmacport/files/6265158/02.configure.cc.txt)
[02.configure.txt](https://github.com/railwaycat/homebrew-emacsmacport/files/6265159/02.configure.txt)

**Comment by cwhatley** [comment]
Also, I should mention that I'm on XCode 12.4 (12D4e) on Big Sur 11.3 beta 5.

**Comment by railwaycat** [comment]
Right, `checking for Carbon/Carbon.h... no` is the key part. After that the configure checking turns to X libs instead of mac. 

I never test on Big Sur Beta so not sure about the latest changes. Can you try to `find /Library/Developer -name Carbon.h` to see if `Carbon.h` header exists?

**Comment by cwhatley** [comment]
There was no Carbon.h anywhere in /Library/Developer though it is found under XCode.app which seems odd.

I ended up blowing away /Library/Developer/CommandLineTools and reinstalling. Problem solved.

What was very odd was when I tried building emacs-mac outside of the brew universe and their shim scripts for the compiler and other dev tools, Carbon.h was found via the XCode.app install. The brew shims lock things down to only use /Library/Developer/CommandLineTools and block visibility into XCode.app install.
